keps/sig-storage/661-statefulset-volume-resize/README.md

```diff
@@ -0,0 +1,1112 @@
+    * If the roll out strategy is burst (not monotonic), continue.
+
+```
+for pvcTemplate in statefulset {
+  statefulset.status.pvcTemplateStatus[pvcTemplate.name].readyReplicas = 0
+}
+
+for replica in replicas {
+  // this is part of the existing replicas loop in the StatefulSet reconciliation process (https://github.com/kubernetes/kubernetes/blob/d89d5ab2680bc74fe4487ad71e514f4e0812d9ce/pkg/controller/statefulset/stateful_set_control.go#L397)
+  // ...
+  if getPodRevision(replica) == statefulset.revision && replica.Status.Phase == PodRunning && replica.DeletionTimestamp == nil {
+    for pvcTemplate in statefulset {
+      pvc, err = getPodPVC(replica, pvcTemplate)
+      if err != nil {
+        // this is either a transient error, or an error that shouldn't happen, because the pod is running and the PVC should exist
+        recordError(statefulset, err)
+        return err
+      }
+
+      if requestStorage(pvcTemplate) != requestStorage(pvc) {
+        err = patchPVC(pvc, requestStorage(pvcTemplate))
+        if err != nil {
+          recordError(statefulset, err)
+          if isTransientError(err) {
+            return err
+          } else if monotonic {
+            return nil
+          }
+        }
+      }
+
+      if requestStorage(pvcTemplate) == capacityStorage(pvc) {
+        statefulset.status.pvcTemplateStatus[pvcTemplate.name].readyReplicas++
+      }
+    }
+  }
+  // ...
+}
+
+for pvcTemplate in statefulset {
+  if statefulset.status.pvcTemplateStatus[pvcTemplate.name].readyReplicas == statefulset.spec.replicas {
+    statefulset.status.pvcTemplateStatus[pvcTemplate.name].finishedReconciliationGeneration = statefulset.generation
+  }
+}
+```
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+- E2E tests for volume expansion of PVC associated with statefulsets.
+- Ensure statefulset controller revision works properly after resize.
+- Test resize failure recovery after [KEP 1790](../1790-recover-resize-failure/README.md) has been implemented.
+
+* We'll need to make sure that while the feature flag is disabled
```

Comment:
Can there be a way to disable it on a per-STS basis? We have a use case for some of our STS where we want different replicas to have different size volumes.